Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/parsers/BaseDatePeriodParser.java

```diff
@@ -1002,6 +1005,14 @@ private DateTimeResolutionResult mergeTwoTimePoints(String text, LocalDateTime r
 
             pr1 = dateContext.processDateEntityParsingResult(pr1);
             pr2 = dateContext.processDateEntityParsingResult(pr2);
+
+            // When the case has no specified year, we should sync the future/past year due to invalid date Feb 29th.
+            if (dateContext.isEmpty() && (DateContext.isFeb29th((LocalDateTime)((DateTimeResolutionResult)pr1.getValue()).getFutureValue()) ||
+                    DateContext.isFeb29th((LocalDateTime)((DateTimeResolutionResult)pr2.getValue()).getFutureValue()))) {
+                HashMap<String, DateTimeParseResult> parseResultHashMap = dateContext.syncYear(pr1, pr2);
```

Comment:
Nitpick: I'd add a blank line here, before the declaration of the map, to make readability of the code easier.